### PR TITLE
nginx: add defaultPort and defaultSSLPort options

### PR DESCRIPTION
**Pull request by admiral0**
###### Description of changes

These 2 new options allow you to specify the default ports for Nginx for all the vhosts - allowing for example to use openvpn's portshare feature. 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 1]
```diff
@@ -446,6 +446,24 @@ in
...
+          Default SSL port for the server
+        ";
+      };
+
+      defaultPort = mkOption {
+        type = types.port;
+        default = 80;
+        example = literalExpression ''80'';
+        description = "
+          Default port for the server
```
Comment (lovesegfault) [inline nixos/modules/services/web-servers/nginx/default.nix:463]: Perhaps this wording helps make it beyond-obvious by contrasting with "SSL" in the docs for `defaultSSLPort`?
```suggestion
          Default HTTP port for the server
```

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 2]
```diff
@@ -446,6 +446,24 @@ in
         ";
       };
 
+      defaultSSLPort = mkOption {
+        type = types.port;
+        default = 443;
+        example = literalExpression ''443'';
```
Comment (lovesegfault) [inline nixos/modules/services/web-servers/nginx/default.nix:452]: We only use `''` when you need multiline strings with contextual indentation

```suggestion
        example = literalExpression "443";
```

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 3]
```diff
@@ -446,6 +446,24 @@ in
...
+        description = "
+          Default SSL port for the server
+        ";
+      };
+
+      defaultPort = mkOption {
+        type = types.port;
+        default = 80;
+        example = literalExpression ''80'';
```
Comment (lovesegfault) [inline nixos/modules/services/web-servers/nginx/default.nix:461]:
```suggestion
        example = literalExpression "8080";
```

**Review by lovesegfault — APPROVED** [review]
Left a couple of nits, but LGTM.

Tagged some more relevant folks for a second look.

File: nixos/modules/services/web-servers/nginx/default.nix (hunk 2 above)
Comment (pennae) [reply nixos/modules/services/web-servers/nginx/default.nix:452]: using `literalExpression` isn't necessary here anyway, numeric literals can just be written as is.

**Comment by ck3d** [comment]
Please squash the commits and remove the examples.

**Comment by dali99** [comment]
superseded by #198529

**Pull request closed without merging**